sql/query/update_test.go

```diff
@@ -77,22 +77,50 @@ func TestUpdateStmt(t *testing.T) {
 	}
 
 	t.Run("with arrays", func(t *testing.T) {
-		db, err := genji.Open(":memory:")
-		require.NoError(t, err)
-		defer db.Close()
-
-		res, err := db.Query(`
-			CREATE TABLE foo;
-			INSERT INTO foo (a) VALUES ([1, 0, 0]), ([2, 0]);
-			UPDATE foo set b = 0;
-			SELECT * FROM foo; 
-		`)
-		require.NoError(t, err)
-		defer res.Close()
-
-		var buf bytes.Buffer
-		err = document.IteratorToJSONArray(&buf, res)
-		require.NoError(t, err)
-		require.JSONEq(t, `[{"a": [1, 0, 0], "b": 0}, {"a": [2, 0], "b": 0}]`, buf.String())
+		tests := []struct {
+			name     string
+			query    string
+			fails    bool
+			expected string
+			params   []interface{}
+		}{
+			{"SET / No cond add field ", `UPDATE foo set b = 0`, false, `[{"a": [1, 0, 0], "b": 0}, {"a": [2, 0], "b": 0}]`, nil},
+			{"SET / No cond / with dot notation at existing index only", `UPDATE foo SET a[2] = 10`, false, `[{"a": [1, 0, 10]}, {"a": [2, 0]}]`, nil},
```

Comment:
We should use `path` instead of `dot notation` in these. No need to rename other tests that still have that naming.